llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -2775,6 +2775,23 @@ BasicBlock *InnerLoopVectorizer::createVectorizedLoopSkeleton(
   return LoopVectorPreHeader;
 }
 
+static bool isValueIncomingFromBlock(BasicBlock *ExitingBB, Value *V,
+                                     Instruction *UI) {
+  PHINode *PHI = dyn_cast<PHINode>(UI);
```

Comment:
```suggestion
  PHINode *PHI = cast<PHINode>(UI);
```

No need for a separate assert. Might be better to make the parameter a PHINode * instead to make the interface clearer (that this function only works if UI is a phi).

Reply:
I think the reason why it was a dyn_cast prior to my patch is because cast doesn't give as detailed an error message, which is why I kept the dyn_cast here. Having said that, this code won't last long anyway since https://github.com/llvm/llvm-project/pull/112147 removes fixupIVUsers.